docs/source/openapi.json

```diff
@@ -3317,6 +3317,370 @@
+        ],
+        "parameters": [
+          {
+            "$ref": "#/components/parameters/RequiredDataset"
+          },
+          {
+            "$ref": "#/components/parameters/RequiredConfig"
+          },
+          {
+            "$ref": "#/components/parameters/RequiredSplit"
+          },
+          {
+            "name": "where",
+            "in": "query",
+            "description": "The filter query.",
+            "required": true,
+            "schema": {
+              "type": "string"
+            },
+            "examples": {
+              "Age = 30": {
+                "summary": "filter the rows where the 'Age' column equals to 30",
+                "value": "Age = 30"
+              }
+            }
+          },
+          {
+            "name": "offset",
+            "in": "query",
+            "description": "The offset of the returned rows.",
+            "schema": {
+              "type": "integer",
+              "default": 0,
+              "minimum": 0
+            },
+            "examples": {
+              "0": {
+                "summary": "from the beginning",
+                "value": 0
+              },
+              "100": {
+                "summary": "ignore the first 100 results",
+                "value": 100
+              }
+            }
+          },
+          {
+            "name": "length",
+            "in": "query",
+            "description": "The maximum number of returned rows",
+            "schema": {
+              "type": "integer",
+              "default": 100,
+              "minimum": 0,
+              "maximum": 100
+            },
+            "examples": {
+              "100": {
+                "summary": "up to 100 rows in the response",
+                "value": 100
+              }
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The <a href='https://huggingface.co/docs/datasets/about_dataset_features'>features</a>, and the list of rows that match the filter query. Bytes columns are not supported at the moment, and their content will be 'null'.",
+            "headers": {
+              "Cache-Control": {
+                "$ref": "#/components/headers/Cache-Control"
+              },
+              "Access-Control-Allow-Origin": {
+                "$ref": "#/components/headers/Access-Control-Allow-Origin"
+              }
+            },
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/PaginatedResponse"
+                },
+                "examples": {
+                  "Filter on a text dataset (julien-c/titanic-survival)": {
+                    "summary": "The first 2 rows where the 'Age' column equals to 30 (where=Age=30&length=2).",
+                    "description": "Try with: https://datasets-server.huggingface.co/filter?dataset=julien-c/titanic-survival&config=default&split=train&where=Age=30&length=2",
```

Comment:
could you add some more examples with different queries (different operators + different column types, for example)?

Reply:
There are 4 examples now on:
- float equality
- string equality
- float inequality
- logical and with integer equality/inequality